plugins/importexport/csv/CSVImportExportPlugin.inc.php

```diff
@@ -82,190 +82,414 @@ function executeCLI($scriptName, &$args) {
+
+		$user = $userDao->getByUsername($username);
+		if (!$user) {
+			echo __('plugins.importexport.csv.unknownUser', ['username' => $username]) . "\n";
+			exit(1);
+		}
+
+		$file = null;
+
+		try {
+			$file = new SplFileObject($filename, 'r');
+		}
+		catch (Exception $e) {
+			echo $e->getMessage();
+			exit(1);
 		}
 
-		$data = file($filename);
+		$file->setFlags(SplFileObject::READ_CSV);
+
+		if ($file->eof()) {
+			echo __('plugins.importexport.csv.fileDoesNotExist', ['filename' => $filename]) . "\n";
+			exit;
+		}
+
+		$expectedHeaders = [
+			'pressPath',
+			'authorString',
+			'title',
+			'abstract',
+			'seriesPath',
+			'year',
+			'isEditedVolume',
+			'locale',
+			'filename',
+			'doi',
+			'keywords',
+			'subjects',
+			'bookCoverImage',
+			'bookCoverImageAltText',
+			'categories',
+		];
+		$headers = $file->fgetcsv();
+
+		$missingHeaders = array_diff($expectedHeaders, $headers);
+
+		if (count($missingHeaders)) {
+			echo __('pugin.importexport.csv.missingHeadersOnCsv', ['missingHeaders' => $missingHeaders]);
+			exit(1);
+		}
+
+		// Imports
+		import('lib.pkp.classes.submission.SubmissionFile'); // constants.
+		import('lib.pkp.classes.file.FileManager');
+		import('lib.pkp.classes.core.Core');
+		import('classes.file.PublicFileManager');
+
+		// DAOs
+		$pressDao = Application::getContextDAO();
+		$genreDao = DAORegistry::getDAO('GenreDAO'); /* @var $genreDao GenreDAO */
+		$userGroupDao = DAORegistry::getDAO('UserGroupDAO'); /* @var $userGroupDao UserGroupDAO */
+		$submissionDao = DAORegistry::getDAO('SubmissionDAO'); /* @var $submissionDao SubmissionDAO */
+		$publicationDao = DAORegistry::getDAO('PublicationDAO'); /* @var $publicationDao PublicationDAO */
+		$seriesDao = DAORegistry::getDAO('SeriesDAO'); /* @var $seriesDao SeriesDAO */
+		$authorDao = DAORegistry::getDAO('AuthorDAO'); /* @var $authorDao AuthorDAO */
+		$publicationFormatDao = DAORegistry::getDAO('PublicationFormatDAO'); /* @var $publicationFormatDao PublicationFormatDAO */
+		$publicationDateDao = DAORegistry::getDAO('PublicationDateDAO'); /* @var $publicationDateDao PublicationDateDAO */
+		$submissionFileDao = DAORegistry::getDAO('SubmissionFileDAO'); /* @var $submissionFileDao SubmissionFileDAO */
+		$categoryDao = DAORegistry::getDAO('CategoryDAO'); /* @var $categoryDao CategoryDAO */
+
+		// Static variables
+		$fileManager = new FileManager();
+		$publicFileManager = new PublicFileManager();
+		$dirNames = Application::getFileDirectories();
+		$format = trim($dirNames['context'], '/') . '/%d/' . trim($dirNames['submission'], '/') . '/%d';
+		$allowedFileTypes = ['gif', 'jpg', 'png', 'webp'];
+
+		/** @var \PKP\services\PKPFileService */
+		$fileService = Services::get('file');
+		/** @var \PKP\services\PKPPublicationService */
+		$publicationService = Services::get('publication');
+
+		// Cache variables
+		$genres = [];
+		$authorGroups = [];
+		$series = [];
+		$cachedCategories = [];
+		$presses = [];
```

Comment:
As arrays in PHP are not passed by reference, if you want to initialize all together, it should be ok.

```suggestion
		$genres = $authorGroups = $series = $cachedCategories = $presses = [];
```

Reply:
Nice suggestion! I'm working on your round now :laughing: